### ISSUE TITLE
Pinning stops receiving blocks and requires manual reconnect constantly

### ISSUE THREAD

**Issue by VictorBjelkholm** [ipfs/go-ipfs#5360]
#### Version information:

Local

```
go-ipfs version: 0.4.17-
Repo version: 7
System version: amd64/linux
Golang version: go1.10.3
```

Remote

```
go-ipfs version: 0.4.13-
Repo version: 6
System version: amd64/linux
Golang version: go1.9.2
```

#### Type:

Bug?

#### Description:

I have a archive I'm trying to transfer to a remote machine (`QmYMfSnEREWgPAbpbNByyE7xrZgbY6DrGxJxt3Do6c9DEt`). I have a daemon locally and remotely running fine without any crashes. However, after starting the pinning process, it gets hung and stops downloading new blocks. This usually takes an hour or something like that to happen.

Not until I connect them manually with a `swarm connect` do the pinning continue. Even then, after ~1 hour, it hangs again and requires a `swarm connect` again to continue. This repeats until the pinning is done.

**Comment by VictorBjelkholm** [comment]
Made a short demonstration of it in action, observe the bottom-right pane where it doesn't continue until I run `swarm connect` in the bottom-left pane https://ipfs.io/ipfs/QmNyKRL1cTNB3o3JNYRKCihQY1YtSpJf2raphPRnaur5wT/pinning-hang.mp4

**Comment by Stebalien** [comment]
Closing in favor of https://github.com/ipfs/go-ipfs/issues/5183. It may not be the same issue, but it looks like it.
